src/log/access_log.cc

```diff
@@ -56,11 +59,12 @@ typedef struct {
     hash_link hash;
     int n;
 } fvdb_entry;
-static hash_table *via_table = NULL;
+using fvdb_table_t = std::unordered_map<SBuf,int>;
```

Comment:
I can only repeat the argument I have given, but I will try to spell it out better: This storage may grow without bounds, forever, causing Squid to run out of RAM. We should account for that memory usage for hopefully obvious reasons. The best way to account for that here is to pool these allocations.

This primary argument outweighs (by far!) any other arguments against pooling that I can imagine. If you have arguments for why we should _not_ pool here, _and_  you think those arguments outweigh the spelled out argument above and the general "pool allocations" principle, please provide those arguments. I would be happy to be corrected as it would result in slightly simpler code here and possibly even fewer change requests in the future. Your response did not contain any such arguments AFAICT.

----

This is not particularly relevant to this change request, but since you mentioned SBuf and value_type allocations, I decided to add this to avoid further misunderstanding:

* AFAICT, std::unordered_map does not allocate individual SBufs or even value_types. It allocates hash buckets. The size of each bucket allocation probably depends on many factors, and I would expect it to vary over the map lifetime (if the hash load factor changes enough), but I do not know the details.
* Memory pools optimize frequent deallocations (their absolute size is of secondary importance -- it is more about their relationship with the system allocator page size). Memory pools also track memory usage. Both features are important; some memory pools uses are more about the former feature and some are more about the latter.

Reply:
That's a really good point, thanks for raising it - especially the one about tracking memory.
The allocator is for value_type, aka std::pair<SBuf, uint64_t>

Done in 46006f6b5

Reply:
> The allocator is for value_type

That is how it _looks_, but that is not what it actually allocates.